src/imputors/substitute.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    Substitute(; statistic=nothing)
+    Substitute(; robust=true, weights=nothing)
```

Comment:
I wonder if we couldn't get ride of `robust` and just introduce `default_statistics_median` and `default_statistics_mean`

and always pass in the weights to all statistics.

Reply:
Requiring all functions to take weights seems like a bit of a nuisance as changing this to `mean ∘ skipmissing` is really convenient.

Reply:
Fair 
Fir future thought

Reply:
https://github.com/invenia/Impute.jl/issues/79